platforms/substrate/charts/substrate-genesis/templates/genesis.yaml

```diff
@@ -195,6 +197,29 @@ spec:
             kubectl create configmap "substrate-genesis" --namespace {{ .Release.Namespace }} --from-file=genesis="${MOUNT_PATH}/genesis_base64" --from-file=sudoKeygenOutput="${MOUNT_PATH}/sudoKeygenOutput.json"
           fi
           echo "COMPLETED!"
+{{- if eq .Values.global.vault.type "hashicorp" }}
+          # Initialize the token
+          . /scripts/bevel-vault.sh
+          vaultBevelFunc "init"
+          vault_secret_key="${VAULT_SECRET_ENGINE}/${VAULT_SECRET_PREFIX}/genesis"
+          vaultBevelFunc "readJson" "$vault_secret_key"
+
+          # The vault CLI is required for this job as the genesis file is too large to be passed in via a vault API call
+          echo "Installing Vault CLI"
+          curl -O -L https://releases.hashicorp.com/vault/1.7.1/vault_1.7.1_linux_amd64.zip
+          unzip vault_1.7.1_linux_amd64.zip
+          {{- if eq $.Values.node.image "docker.io/paritytech/substrate-playground-template-node-template" }}
+          export PATH=$PATH:~/workspace
+          {{- else }}
+          mv vault /bin
+          {{- end }}
+          vault --version
+          # Save the generated keys to VAULT
+          if [ "$SECRETS_AVAILABLE" == "no" ]; then
+            vault_secret_key="${VAULT_SECRET_ENGINE}/${CERTS_SECRET_PREFIX}/genesis"
+            vault kv put "$vault_secret_key" genesis=@"${MOUNT_PATH}/genesis_base64" sudo_details=@"${MOUNT_PATH}/sudoKeygenOutput.json"
```

Comment:
why are we downloading and using vault cli ? why not use the bevel vault script ?